clients/ui/frontend/src/app/pages/modelCatalog/components/ModelCatalogActiveFilters.tsx

```diff
@@ -30,57 +34,99 @@ const ModelCatalogActiveFilters: React.FC<ModelCatalogActiveFiltersProps> = ({ f
     }
   };
 
-  const getFilterLabel = (filterKey: ModelCatalogStringFilterKey, value: string): string => {
-    switch (filterKey) {
-      case ModelCatalogStringFilterKey.PROVIDER: {
-        return isEnumMember(value, ModelCatalogProvider)
-          ? MODEL_CATALOG_PROVIDER_NAME_MAPPING[value]
-          : value;
-      }
-      case ModelCatalogStringFilterKey.LICENSE: {
-        return isEnumMember(value, ModelCatalogLicense)
-          ? MODEL_CATALOG_LICENSE_NAME_MAPPING[value]
-          : value;
+  /**
+   * Gets the display label for a filter value based on the filter key type
+   */
+  const getFilterLabel = (filterKey: ModelCatalogFilterKey, value: string | number): string => {
+    // Handle string filter keys
+    if (isEnumMember(filterKey, ModelCatalogStringFilterKey)) {
+      const valueStr = String(value);
+      switch (filterKey) {
+        case ModelCatalogStringFilterKey.PROVIDER: {
+          return isEnumMember(valueStr, ModelCatalogProvider)
+            ? MODEL_CATALOG_PROVIDER_NAME_MAPPING[valueStr]
+            : valueStr;
+        }
+        case ModelCatalogStringFilterKey.LICENSE: {
+          return isEnumMember(valueStr, ModelCatalogLicense)
+            ? MODEL_CATALOG_LICENSE_NAME_MAPPING[valueStr]
+            : valueStr;
+        }
+        case ModelCatalogStringFilterKey.TASK: {
+          return isEnumMember(valueStr, ModelCatalogTask)
+            ? MODEL_CATALOG_TASK_NAME_MAPPING[valueStr]
+            : valueStr;
+        }
+        case ModelCatalogStringFilterKey.LANGUAGE: {
+          return isEnumMember(valueStr, AllLanguageCode) ? AllLanguageCodesMap[valueStr] : valueStr;
+        }
+        default:
+          return valueStr;
       }
-      case ModelCatalogStringFilterKey.TASK: {
-        return isEnumMember(value, ModelCatalogTask)
-          ? MODEL_CATALOG_TASK_NAME_MAPPING[value]
-          : value;
+    }
+
+    // Handle number filter keys
+    if (isEnumMember(filterKey, ModelCatalogNumberFilterKey)) {
+      switch (filterKey) {
+        case ModelCatalogNumberFilterKey.MIN_RPS:
+          return `≥${value} requests/sec`;
+        default:
+          return String(value);
       }
-      case ModelCatalogStringFilterKey.LANGUAGE: {
-        return isEnumMember(value, AllLanguageCode) ? AllLanguageCodesMap[value] : value;
+    }
+
+    // Handle latency field names
+    if (isLatencyFieldName(filterKey)) {
+      const parsed = parseLatencyFieldName(filterKey);
+      if (parsed) {
+        return `${parsed.metric} ${parsed.percentile}: ≤${value}ms`;
       }
-      default:
-        return value;
+      return `${filterKey}: ≤${value}ms`;
     }
+
+    return String(value);
   };
 
   return (
     <>
       {filtersToShow.map((filterKey) => {
-        // Only process string filter keys that are arrays
-        if (!isEnumMember(filterKey, ModelCatalogStringFilterKey)) {
+        const filterValue = filterData[filterKey as keyof typeof filterData];
```

Comment:
This type assertion isn't necessary, `filterKey` is already a valid type to use as a key of `filterData`.